pilot/pkg/proxy/envoy/v1/gateway.go

```diff
@@ -75,23 +70,14 @@ func buildPhysicalGatewayListener(
 	server *routing.Server,
 ) *Listener {
 
-	//TODO: at this given time the rds can only be 1 service port
-	// should rds be a collection of service ports?
-	var rdsName string
-	if len(allServicesInTheMesh) > 0 {
-		if len(allServicesInTheMesh[0].Ports) > 0 {
-			rdsName = strconv.Itoa(allServicesInTheMesh[0].Ports[0].Port)
-		}
-	}
-
 	opts := buildHTTPListenerOpts{
 		mesh:             mesh,
 		node:             node,
 		instances:        instancesOnThisNode,
 		routeConfig:      nil,
 		ip:               WildcardAddress,
 		port:             int(server.Port.Number),
-		rds:              rdsName,
+		rds:              strconv.Itoa(int(server.Port.Number)),
```

Comment:
@ZackButcher / @rshriram we've gone back to this - we want the rds to be the port that we are exposing on the envoy. Seems to make the most sense.

Hopefully that will work.